File: crm-web/src/types/emailTemplate.ts
export interface EmailTemplate {
  id: string;
  name: string;
  subject: string;
  bodyHtml: string;
  bodyText?: string;
  category?: string;
  availableVariables?: string;
  usageCount: number;
  openRate?: number;
  clickRate?: number;
  isActive: boolean;
  createdBy?: { id: string; fullName: string };
  createdAt: string;
  updatedAt: string;
}

export interface EmailTemplateListItem {
  id: string;
  name: string;
  subject: string;
  category?: string;
  isActive: boolean;
  usageCount: number;
  openRate?: number;
  clickRate?: number;
  createdByName?: string;
  updatedAt?: string;
}

export interface CreateEmailTemplateRequest {
  name: string;
  subject: string;
  bodyHtml: string;
  bodyText?: string;
  category?: string;
  availableVariables?: string;
}

export interface UpdateEmailTemplateRequest {
  name?: string;
  subject?: string;
  bodyHtml?: string;
  bodyText?: string;
  category?: string;
  availableVariables?: string;
}

export const EMAIL_CATEGORIES = [
  { value: 'commercial', label: 'Commercial' },
  { value: 'marketing', label: 'Marketing' },
  { value: 'support', label: 'Support' },
  { value: 'notification', label: 'Notification' },
  { value: 'relance', label: 'Relance' },
  { value: 'other', label: 'Autre' },
];

export const EMAIL_CATEGORY_CONFIG: Record<string, { label: string; color: string }> = {
  commercial: { label: 'Commercial', color: 'blue' },
  marketing: { label: 'Marketing', color: 'purple' },
  support: { label: 'Support', color: 'cyan' },
  notification: { label: 'Notification', color: 'orange' },
  relance: { label: 'Relance', color: 'green' },
  other: { label: 'Autre', color: 'default' },
};

File: crm-backend/crm-communication/src/main/java/com/opticrm/communication/entity/EmailTemplate.java
package com.opticrm.communication.entity;

import com.opticrm.security.entity.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "email_templates")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class EmailTemplate {

    @Id
    @UuidGenerator
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "subject", nullable = false, length = 255)
    private String subject;

    @Column(name = "body_html", nullable = false, columnDefinition = "TEXT")
    private String bodyHtml;

    @Column(name = "body_text", columnDefinition = "TEXT")
    private String bodyText;

    @Column(name = "category", length = 50)
    private String category;

    @Column(name = "available_variables", columnDefinition = "jsonb")
    private String availableVariables;

    @Column(name = "usage_count")
    @Builder.Default
    private Integer usageCount = 0;

    @Column(name = "open_rate", precision = 5, scale = 2)
    private BigDecimal openRate;

    @Column(name = "click_rate", precision = 5, scale = 2)
    private BigDecimal clickRate;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_id")
    private User createdBy;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private Instant updatedAt;
}
